kuzzleio/kuzzle
# KZL 290 - Add fixtures for securities

## What does this PR do ?

This PR allow to load fixtures for users, profiles and roles at Kuzzle startup.  
The roles and profiles are loaded with `createOrReplace` and the users are first deleted with `mDeleteUsers` and then created with `createUser`.  

Documentation PR: https://github.com/kuzzleio/documentation/pull/521

The securities fixtures must have the following format: 

```json
{
  "roles": {
    "driver": {
      "controllers": {
        "auth": {
          "actions": {
            "checkToken": true,
            "getCurrentUser": true,
            "getMyRights": true,
            "login": true
          }
        },
        "document": {
          "actions": {
            "*": true
          }
        }
      }
    }
  },
  "profiles": {
    "driver": {
      "policies": [
        {
          "roleId": "driver",
          "restrictedTo": [
            { "index": "nyc-open-data", "collections": ["yellow-taxi"] }
          ]
        }
      ]
    }
  },
  "users": {
    "bcalhoun": {
      "content": {
        "profileIds": ["driver"]
      },
      "credentials": {
        "local": {
          "username": "bcalhoun",
          "password": "blueshift"
        }
      }
    }
  }
}
```

### How should this be manually tested?

  - Step 1 : Start only `elasticsearch` and `redis` with docker-compose (don't forget to expose the ports 9200 and 6379 respectively)
  - Step 2 : Run Kuzzle from the commande line: `./bin/kuzzle start --fixtures test/mocks/securities.json`
  - Step 3 : Check the roles, profiles and users presence in the admin console

### Other changes

  - Move mappings and fixtures loading to Janitor
  - Fix node inspector in docker-compose
  - Fix #1184 : Refactor a little the method `load` in the repositories to reject with an explicit `NotFoundError`. Some methods (`persistUser` generator from SecurityController and `load` from PluginRepository) needed to return `null` instead of rejecting with a 404 so I added an optional `returnNull` options to the `loadOneFromDatabase` method.
